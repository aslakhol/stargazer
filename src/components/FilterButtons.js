import React from 'react';
import { Button, ButtonGroup } from 'reactstrap';

const FilterButtons = (props) => {
  const { onCheck, filter } = props;
  const filterOptions = ['Male', 'Female', 'Hermaphrodite', 'N/A'];
  return (
    <ButtonGroup>
      {filterOptions.map(option => (
        <Button color="primary" onClick={() => onCheck(option)} active={!filter.includes(option)} key={option}>{option}</Button>
      ))}
    </ButtonGroup>
  );
};

export default FilterButtons;
