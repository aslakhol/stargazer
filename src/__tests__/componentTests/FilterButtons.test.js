import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import FilterButtons from '../../components/FilterButtons';

describe('<FilterButtons />', () => {
  it('renders the component', () => {
    const wrapper = shallow(<FilterButtons filter={[]} />);
    const component = wrapper.dive();

    expect(toJson(component)).toMatchSnapshot();
  });

  it('renders 4 buttons', () => {
    const wrapper = shallow(<FilterButtons filter={[]} />);

    expect(wrapper.find('Button')).toHaveLength(4);
  });

  describe('onClick', () => {
    it('calls onCheck', () => {
      const mockOnCheck = jest.fn();
      const wrapper = shallow(<FilterButtons filter={[]} onCheck={mockOnCheck} />);

      const button = wrapper.find('Button').first();
      button.simulate('click');
      expect(mockOnCheck.mock.instances.length).toBe(1);
    });
  });
});
