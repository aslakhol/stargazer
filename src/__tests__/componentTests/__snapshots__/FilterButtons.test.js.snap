// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FilterButtons /> renders the component 1`] = `
<div
  className="btn-group"
  role="group"
>
  <Button
    active={true}
    color="primary"
    key="Male"
    onClick={[Function]}
    tag="button"
  >
    Male
  </Button>
  <Button
    active={true}
    color="primary"
    key="Female"
    onClick={[Function]}
    tag="button"
  >
    Female
  </Button>
  <Button
    active={true}
    color="primary"
    key="Hermaphrodite"
    onClick={[Function]}
    tag="button"
  >
    Hermaphrodite
  </Button>
  <Button
    active={true}
    color="primary"
    key="N/A"
    onClick={[Function]}
    tag="button"
  >
    N/A
  </Button>
</div>
`;
